src/NRules/NRules.RuleModel/NamedExpressionElement.cs

```diff
@@ -7,10 +7,10 @@ namespace NRules.RuleModel
     /// </summary>
     public class NamedExpressionElement : ExpressionElement
     {
-        internal NamedExpressionElement(string name, LambdaExpression expression) 
-            : base(expression)
+        internal NamedExpressionElement(NamedLambdaExpression namedLambdaExpression) 
```

Comment:
It's much better to pass name and expression separately here, to not couple to named expression representation. So, I would revert the change in this file.